src/main/java/org/openstreetmap/atlas/proto/adapters/ProtoPackedTagStoreAdapter.java

```diff
@@ -0,0 +1,166 @@
+package org.openstreetmap.atlas.proto.adapters;
+
+import java.lang.reflect.Field;
+
+import org.openstreetmap.atlas.exception.CoreException;
+import org.openstreetmap.atlas.geography.atlas.packed.PackedTagStore;
+import org.openstreetmap.atlas.proto.ProtoIntegerArrayOfArrays;
+import org.openstreetmap.atlas.proto.ProtoPackedTagStore;
+import org.openstreetmap.atlas.proto.ProtoSerializable;
+import org.openstreetmap.atlas.proto.converters.ProtoIntegerArrayOfArraysConverter;
+import org.openstreetmap.atlas.utilities.arrays.IntegerArrayOfArrays;
+
+import com.google.protobuf.InvalidProtocolBufferException;
+
+/**
+ * Implements the {@link ProtoAdapter} interface to connect {@link PackedTagStore} and
+ * {@link ProtoPackedTagStore}.
+ *
+ * @author lcram
+ */
+
+public class ProtoPackedTagStoreAdapter implements ProtoAdapter
+{
+    /*
+     * PackedTagStore does not provide an interface for setting its arrays directly. This class's
+     * implementation uses reflection to side-step the issue. Unlike the other PackedAtlas field
+     * classes, there is no nice converter for the ProtoPackedTagStore (since it cannot be rebuilt
+     * through its public API).
+     */
+
+    private static final ProtoIntegerArrayOfArraysConverter CONVERTER = new ProtoIntegerArrayOfArraysConverter();
+
+    @Override
+    public ProtoSerializable deserialize(final byte[] byteArray)
+    {
+        ProtoPackedTagStore protoStore = null;
+        try
+        {
+            protoStore = ProtoPackedTagStore.parseFrom(byteArray);
+        }
+        catch (final InvalidProtocolBufferException exception)
+        {
+            throw new CoreException("Error encountered while parsing protobuf bytestream",
+                    exception);
+        }
+        final PackedTagStore store = new PackedTagStore();
+
+        final ProtoIntegerArrayOfArrays protoKeyArray = protoStore.getKeys();
+        final ProtoIntegerArrayOfArrays protoValueArray = protoStore.getValues();
+
+        final IntegerArrayOfArrays keyArray = CONVERTER.convert(protoKeyArray);
+        final IntegerArrayOfArrays valueArray = CONVERTER.convert(protoValueArray);
+        final Long index = protoStore.getIndex();
+
+        Field keyArrayField = null;
+        Field valueArrayField = null;
+        Field indexField = null;
+
+        try
+        {
+            keyArrayField = store.getClass().getDeclaredField(PackedTagStore.FIELD_KEYS);
+            keyArrayField.setAccessible(true);
+            keyArrayField.set(store, keyArray);
+        }
+        catch (NoSuchFieldException | SecurityException | IllegalArgumentException
```

Comment:
All the exceptions get handled the same way - there is no difference what kind of Exception it is, we can't recover from it. Might be a good spot to just catch `Exception`.